python/packages/core/agent_framework/_workflows/_handoff.py

```diff
@@ -0,0 +1,1388 @@
+        if not conversation:
+            raise ValueError("Handoff workflow requires non-empty conversation before requesting user input.")
+        request = HandoffUserInputRequest(
+            conversation=list(conversation),
+            awaiting_agent_id=self._starting_agent_id,
+            prompt=self._prompt,
+        )
+        request.source_executor_id = self.id
+        await ctx.send_message(request, target_id=self._request_executor_id)
+
+    @handler
+    async def resume_from_user(
+        self,
+        response: RequestResponse[HandoffUserInputRequest, Any],
+        ctx: WorkflowContext[_ConversationWithUserInput],
+    ) -> None:
+        # Reconstruct full conversation with new user input
+        conversation = list(response.original_request.conversation)
+        user_messages = _as_user_messages(response.data)
+        conversation.extend(user_messages)
+
+        # Send full conversation back to coordinator (not trimmed)
+        # Coordinator will update its authoritative history and trim for agent
+        message = _ConversationWithUserInput(full_conversation=conversation)
+        # CRITICAL: Must specify target to avoid broadcasting to all connected executors
+        # Gateway is connected to both request_info and coordinator, we want coordinator only
+        await ctx.send_message(message, target_id="handoff-coordinator")
+
+
+def _as_user_messages(payload: Any) -> list[ChatMessage]:
+    if isinstance(payload, ChatMessage):
+        if payload.role == Role.USER:
+            return [payload]
+        return [ChatMessage(Role.USER, text=payload.text)]
+    if isinstance(payload, list):
+        # Check if all items are ChatMessage instances
+        all_chat_messages = all(isinstance(msg, ChatMessage) for msg in payload)  # type: ignore[arg-type]
+        if all_chat_messages:
+            messages: list[ChatMessage] = payload  # type: ignore[assignment]
+            return [msg if msg.role == Role.USER else ChatMessage(Role.USER, text=msg.text) for msg in messages]
+    if isinstance(payload, Mapping):  # User supplied structured data
+        text = payload.get("text") or payload.get("content")  # type: ignore[union-attr]
+        if isinstance(text, str) and text.strip():
+            return [ChatMessage(Role.USER, text=text.strip())]
+    return [ChatMessage(Role.USER, text=str(payload))]  # type: ignore[arg-type]
+
+
+def _default_termination_condition(conversation: list[ChatMessage]) -> bool:
+    """Default termination: stop after 10 user messages to prevent infinite loops."""
+    user_message_count = sum(1 for msg in conversation if msg.role == Role.USER)
+    return user_message_count >= 10
+
+
+class HandoffBuilder:
+    r"""Fluent builder for conversational handoff workflows with triage and specialist agents.
+
+    The handoff pattern models a customer support or multi-agent conversation where:
+    - A **triage/dispatcher agent** receives user input and decides whether to handle it directly
```

Comment:
The term "triage" doesn't show up in the API. Maybe we should have it as "add_triage". Personally, I prefer "coordinator" or "dispatcher" for more general naming.